### DISCUSSION TITLE
Can App Layout be used in NextJs layout file?

### DISCUSSION THREAD

**Discussion by rexwreyes** [cloudscape-design/components#1809] [Q&A] [answered]
When App Layout has you include your content as props how do you get it working on a page that has you include your content as { children } ?  Is this just not possible?

**Comment by Al-Dani** [comment]
Hey!
You can pass children to the content slot of the AppLayout:
```
<AppLayout
   ...
   content={props.children}
/>
```

**Comment by Al-Dani** [comment, marked as answer]
_Posting here for posterity, as it was discussed in more details in a separate request._

App Layout cannot be used in the server part of the `layout` file, because it uses hooks that wouldn't work on a server.
You would have to create a client-side wrapper where you can pass children to the content slot:

```
// client-layout.tsx
"use client";
import { AppLayout } from "@cloudscape-design/components";

export default function ClientLayout(props: { children: React.ReactNode }) {
  return <AppLayout content={props.children} />;
}
```

and use it in the layout:
```
// layout.tsx
...
import ClientLayout from './client-layout';
..
export default function RootLayout({ children }: { children: React.ReactNode }) {
  ....
  return (
    <html lang="en">
      ....
      <body>
        <ClientLayout>{children}</ClientLayout>
      </body>
    </html>
  )
}
```
